### ISSUE TITLE
License info on PyPI contains complete license text

### ISSUE THREAD

**Issue by cmeessen** [nucleic/kiwi#137]
The license info on PyPI contains the complete license text:

![image](https://user-images.githubusercontent.com/14222414/161495661-4e075d0c-b08c-4bd1-950c-f21bc8b84c47.png)

It is common practice to use only the identifier for the license. If I am not wrong, in your case the corresponding [SPDX Identifier](https://spdx.org/licenses/) should be `BSD-3-Clause`.

It would be great if you fix this in PyPI. We are having issues with automated license analysis of our dependencies, because the PyPI license string of the kiwisolver cannot be evaluated properly.

Thanks :)

**Comment by MatthieuDartiailh** [comment]
When I read the specification for this field in PEP 621 (https://peps.python.org/pep-0621/#dependencies-optional-dependencies) I believe what I am doing is legitimate. For your purpose I am wondering if looking at classifiers wouldn't be better.

**Comment by cmeessen** [comment]
Fair enough, that's a good point. We will look into it and fix it. Thanks for the quick response, I will close the issue.

**Comment by sccolbert** [comment]
Please keep in mind, phrases like:

“It would be great if you fix… it’s causing issues with our…”

Are one of the big reasons I left software development. It simply reeks of
entitlement.

First, it presupposes something is wrong with the OSS project, when in this
case the issue is on the user’s side.

Second, even if there was a problem with the OSS project, why should I care
if it causes you issues if you can’t even muster a “please” while expecting
it to be fixed. You use this software for free, and obviously derive value
from it. So at least be humble in your communications.

Third, asking a project to change their license text, after almost a decade
in use, is no small request. It should not be treated so flippantly.

Finally, I assume from your last name that you are not a native English
speaker, so I excuse you for a lot of this. However, native speakers do
this all the time too, so I figured I’d say my thoughts on the matter while
it’s fresh.

**Comment by cmeessen** [comment]
Thank you for your communicating your thoughts. I apologise that my request sounded rude or presumptuous, it was not meant like that. I will consider your points in the future and take more time for better phrasing. My intention was to point out that kiwisolver is doing something that most others are doing in a different way.

I would like to expand on the third point you mention, because I think this one was misunderstood. I did not mean to ask to change the actual license of the project. The license text in [LICENSE](https://github.com/nucleic/kiwi/blob/main/LICENSE) corresponds to the [BDS-3-Clause](https://opensource.org/licenses/BSD-3-Clause). That license is sometimes also called "New" or "Revised" which is the title in the license file. Since the vast majority of package providers on PyPI use a text that mostly corresponds to the SPDX Identifier, I was confused to find the complete license text in there. From my point of view, the project's PyPI page may have benefited from stating only the identifier. Especially, because SPDX identifiers are machine readable. My apologies that I did not communicate this properly.
